optuna_dashboard/artifact/_backend.py

```diff
@@ -58,21 +59,21 @@ def register_artifact_route(
     app: Bottle, storage: BaseStorage, artifact_backend: Optional[ArtifactBackend]
 ) -> None:
     @app.get("/artifacts/<study_id:int>/<trial_id:int>/<artifact_id:re:[0-9a-fA-F-]+>")
-    def proxy_artifact(study_id: int, trial_id: int, artifact_id: str) -> bytes:
+    def proxy_artifact(study_id: int, trial_id: int, artifact_id: str) -> HTTPResponse | bytes:
         if artifact_backend is None:
             response.status = 400  # Bad Request
             return b"Cannot access to the artifacts."
         artifact_dict = _get_artifact_meta(storage, study_id, trial_id, artifact_id)
         if artifact_dict is None:
             response.status = 404
             return b"Not Found"
-        response.set_header("Content-Type", artifact_dict["mimetype"])
-        if artifact_dict.get("encoding"):
-            response.set_header("Content-Encodings", artifact_dict.get("encoding"))
+        headers = {"Content-Type": artifact_dict["mimetype"]}
+        encoding = artifact_dict.get("encoding")
+        if encoding:
+            headers["Content-Encodings"] = encoding
 
-        with artifact_backend.open(artifact_id) as f:
-            body = f.read()
-        return body
+        fp = artifact_backend.open(artifact_id)
+        return HTTPResponse(fp, headers=headers)
```

Comment:
Bottle can accept file-like object. The file-like object is properly closed after yielding all the content.

* https://github.com/bottlepy/bottle/blob/99341ff3791b2e7e705d7373e71937e9018eb081/bottle.py#L2614-L2625
* https://github.com/bottlepy/bottle/blob/99341ff3791b2e7e705d7373e71937e9018eb081/bottle.py#L1084-L1085